workers/unity/Packages/com.improbable.gdk.tools/DevAuthTokenUtils.cs

```diff
@@ -33,18 +43,53 @@ private static void Generate()
 
             try
             {
-                if (Json.Deserialize(receivedMessage).TryGetValue(JsonDataKey, out var jsonData) &&
-                    ((Dictionary<string, object>) jsonData).TryGetValue(TokenSecretKey, out var tokenSecret))
+                var deserializedMessage = Json.Deserialize(receivedMessage);
+                if (deserializedMessage.TryGetValue(JsonDataKey, out var jsonData) &&
+                    ((Dictionary<string, object>) jsonData).TryGetValue(JsonTokenSecretKey, out var tokenSecret))
                 {
                     devAuthToken = (string) tokenSecret;
                 }
+                else
+                {
+                    throw new Exception(
+                        $@"{(deserializedMessage.TryGetValue(JsonErrorKey, out var errorMessage)
+                            ? errorMessage
+                            : string.Empty)}");
+                }
             }
             catch (Exception e)
             {
                 Debug.LogError($"Unable to generate Dev Auth Token. {e.Message}");
-                return;
+                return false;
+            }
+
+            Debug.Log($"Saving token to Player Preferences.");
```

Comment:
Is the log needed?

Reply:
given there's a couple ways of saving out a token (player prefs / .txt), I'd prefer to log where it's being saved

this way we'll be able to see if a user's actually created DAT.txt before complaining it doesn't exist